packages/noya-state/src/reducers/layerReducer.ts

```diff
@@ -261,33 +286,84 @@ export function layerReducer(
       );
 
       if (!symbolMasters) return state;
-
       symbolMasters.symbolID = uuid();
-      const symbolInstance = produce(Models.symbolInstance, (draft) => {
-        draft.do_objectID = uuid();
-        draft.name = name;
-        draft.frame = symbolMasters.frame;
-        draft.style = produce(Models.style, (s) => {
-          s.do_objectID = uuid();
-        });
-        draft.symbolID = symbolMasters.symbolID;
-      });
+
+      const originalFrame = { ...symbolMasters.frame };
+      const symbolsPage = state.sketch.pages[symbolsPageIndex];
+
+      const rect = symbolsPage
+        ? getLeftMostLayerPoint(symbolsPage)
+        : { maxX: 0, minY: 25 };
+
+      symbolMasters.frame.x = rect.maxX + 100;
+      symbolMasters.frame.y = rect.minY;
 
       return produce(state, (draft) => {
         const pages = draft.sketch.pages;
 
         deleteLayers(indexPaths, pages[pageIndex]);
-        addSiblingLayer(pages[pageIndex], indexPaths[0], symbolInstance);
 
         const symbolsPage =
           symbolsPageIndex === -1
             ? createPage(pages, draft.sketch.user, 'Symbols')
             : pages[symbolsPageIndex];
+
+        const symbolInstance = produce(Models.symbolInstance, (draft) => {
+          draft.do_objectID = uuid();
+          draft.name = name;
+          draft.frame = originalFrame;
+          draft.style = produce(Models.style, (s) => {
+            s.do_objectID = uuid();
+          });
+          draft.symbolID = symbolMasters.symbolID;
+        });
+
         symbolsPage.layers = [...symbolsPage.layers, symbolMasters];
+        addSiblingLayer(pages[pageIndex], indexPaths[0], symbolInstance);
 
         draft.selectedObjects = [symbolInstance.do_objectID];
       });
     }
+    case 'duplicateLayer': {
+      const [, id] = action;
+
+      const ids = typeof id === 'string' ? [id] : id;
+
+      const pages = state.sketch.pages;
+      const pagesIndexPaths = findPageLayerIndexPaths(state, (layer) =>
+        ids.includes(layer.do_objectID),
+      ).reverse();
+
+      return produce(state, (draft) => {
+        pagesIndexPaths.forEach((pagesIndexPath) => {
+          const { indexPaths, pageIndex } = pagesIndexPath;
+          const page = pages[pageIndex];
+          const draftPage = draft.sketch.pages[pageIndex];
+
+          indexPaths.forEach((indexPath) => {
+            const layer = Layers.access(page, indexPath);
+            const elem = produce(layer, (layer) => {
+              layer.name = layer.name + ' Copy';
+
+              Layers.visit(layer, (layer) => {
+                if (layer.style) layer.style.do_objectID = uuid();
+                if (layer._class === 'symbolMaster') {
```

Comment:
Not super important but I usually go with `Layers.isSymbolMaster` instead of checking `_class`

Reply:
we are using LayerListItem not Layers it's not working 🤔

Reply:
Do you mean in`LayerList.tsx`? You're right, I don't think we can use these functions there

Reply:
Oh I thought this was a different file I just added this change